internal/types/resource/resource.go

```diff
@@ -142,22 +143,22 @@ var Map = map[string]Type{
 
 // Parent returns the parent type for a given type; if there is no parent, it
 // returns the incoming type
-func Parent(in Type) Type {
-	switch in {
+func (r Type) Parent() Type {
```

Comment:
Will also need to update the test in resource_test.go to update all the function signature changes in here

Reply:
Updated: 25fd85d75f82e9af09e02b84b6efaed41a9f2843